src/frontend/src/optimizer/plan_node/logical_share.rs

```diff
@@ -113,7 +114,21 @@ impl ColPrunable for LogicalShare {
     }
 }
 
-impl ExprRewritable for LogicalShare {}
+impl ExprRewritable for LogicalShare {
+    fn has_rewritable_expr(&self) -> bool {
+        true
+    }
+
+    fn rewrite_exprs(&self, r: &mut dyn ExprRewriter) -> PlanRef {
```

Comment:
Please correct me this is not intended for `Share`, i.e., rewrite `input: RefCell<PlanRef>`.
cc @chenzl25 @st1page.